docs: add post for js-IPFS 0.38.0 release (#306)
ipfs/blog: content/post/70-js-ipfs-0-38.md

---
date: 2019-08-30
url: 70-js-ipfs-0-38
title: js-ipfs 0.38.0 released
author: Alex Potsides
---

> Garbage collection, gossipsub, IPNS support for ipfs resolve, oh my!

```console
$ npm install -g ipfs
```

# 🔦 Highlights

## 🗑️ Garbage collection

> Support for garbage collection has shipped with `0.38.0`! This means that blocks in your IPFS Repo will be removed when gc is run unless they are pinned.

Reviewer: ```suggestion
Support for garbage collection has shipped with `0.38.0`! This means that blocks in your IPFS Repo will be removed once gc runs, unless they are pinned.
```

Author: Not sure I prefer `once`.  Sounds so final, like it's not going to be run again.

Reviewer 2: `whenever gc is run`?

Author: `Whensoever gc is run`?

Author: I kid! I made it `whenever`.

Reviewer: sounds good

This is essential for keeping your repo to a manageable size, but must be run manually for the time being.

Use it with the following command:

```console
$ jsipfs repo gc
removed Qmfoo
removed Qmbar
```

To prevent blocks being collected, pin them:

```console
$ jsipfs pin add Qmbaz
pinned Qmbaz recursively
```

## 💬 Gossipsub enabled by default

Pubsub is now enabled by default with Gossipsub as the default implentation.

More efficient than [Floodsub](https://github.com/libp2p/js-libp2p-floodsub), [Gossipsub](https://github.com/ChainSafe/gossipsub-js) creates an overlay on top of the network of nodes which propagates messages to subscribers instead of broadcasting messages to everyone.

Read more about the design at [`libp2p/specs/pubsub/gossipsub`](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub)

## 📛 IPNS support for `ipfs resolve`

You can now resolve [IPNS](https://docs.ipfs.io/guides/concepts/ipns/) names via `ipfs resolve`, what's more the recursive option is now on by default.

```console
$ jsipfs resolve /ipns/Qmqux
/ipfs/Qmgarply
```

> # 🏗 API Changes

Reviewer 2: Did the config for pubsub change in this release? (the EXPERIMENTAL flag?)

* [@Nashatyrev](https://github.com/Nashatyrev) (1 issue)
* [@olizilla](https://github.com/olizilla) (10 comments)
* [@pruflyos](https://github.com/pruflyos) (1 comment)
* [@rabrux](https://github.com/rabrux) (1 comment)
* [@robertkiel](https://github.com/robertkiel) (1 PR)
* [@rvagg](https://github.com/rvagg) (1 comment)
* [@samparsky](https://github.com/samparsky) (18 PRs)
* [@Stebalien](https://github.com/Stebalien) (5 comments)
* [@tapaswenipathak](https://github.com/tapaswenipathak) (1 PR, 1 comment)
* [@teddycode](https://github.com/teddycode) (1 issue)
* [@terichadbourne](https://github.com/terichadbourne) (1 issue)
* [@vasa-develop](https://github.com/vasa-develop) (1 comment)
* [@vasco-santos](https://github.com/vasco-santos) (18 commits, 11 PRs, 18 comments)
* [@vmx](https://github.com/vmx) (2 commits, 3 PRs, 1 comment)
* [@x5engine](https://github.com/x5engine) (1 comment)
* [@yehia67](https://github.com/yehia67) (2 comments)
* [@yuwiggin](https://github.com/yuwiggin) (1 comment)

# 🙌🏽 Want to contribute?

Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:

> - Check the issues with the `help wanted` label at the Ready column in our waffle board - https://waffle.io/ipfs/js-ipfs?label=help%20wanted

Reviewer 2: ```suggestion
- Check the issues with the `help wanted` label in the [js-ipfs repo](https://github.com/ipfs/js-ipfs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
```

- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
- Join the discussion at http://discuss.ipfs.io/ and help users finding their answers.

> - Join the [⚡️ⒿⓈ Core Dev Team Weekly Sync 🙌🏽](https://github.com/ipfs/team-mgmt/issues/650) and be part of the Sprint action!

Reviewer 2: ```suggestion
- Join the [🚀 IPFS Core Implementations Weekly Sync 🛰](https://github.com/ipfs/team-mgmt/issues/992) and be part of the action!
```

Author: /me copy pasta

Oops.

---
date: 2019-08-30
url: 70-js-ipfs-0-38
title: js-ipfs 0.38.0 released
author: Alex Potsides
---

> Garbage collection, gossipsub, IPNS support for ipfs resolve, oh my!

```console
$ npm install -g ipfs
```

> # 🔦 Highlights

Reviewer 3: Suggest  adding one sentence here with the three main highlights (GC, gossipsub, and resolve). Maybe also worth calling out this is the first release through the new process and we need more early testers??

Author: Done!

This is essential for keeping your repo to a manageable size, but must be run manually for the time being.

Use it with the following command:

```console
$ jsipfs repo gc
removed Qmfoo
removed Qmbar
```

To prevent blocks being collected, pin them:

```console
$ jsipfs pin add Qmbaz
pinned Qmbaz recursively
```

## 💬 Gossipsub enabled by default

Pubsub is now enabled by default with Gossipsub as the default implementation.

More efficient than [Floodsub](https://github.com/libp2p/js-libp2p-floodsub), [Gossipsub](https://github.com/ChainSafe/gossipsub-js) creates an overlay on top of the network of nodes which propagates messages to a subset of subscribers instead of broadcasting messages to everyone.

Read more about the design at [`libp2p/specs/pubsub/gossipsub`](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub)

## 📛 IPNS support for `ipfs resolve`

You can now resolve [IPNS](https://docs.ipfs.io/guides/concepts/ipns/) names via `ipfs resolve`, what's more the recursive option is now on by default.

```console
$ jsipfs resolve /ipns/Qmqux
/ipfs/Qmgarply
```

## 🎁 New relase process & a call for early testers

> This is the first release to follow our new, more predicatable, more rigorous release process. It's outlined in the [release issue template](https://github.com/ipfs/js-ipfs/blob/master/doc/RELEASE_ISSUE_TEMPLATE.md), but in short we have defined steps, with an addional period for internal & external testing and gathering feedback on the changes.

Reviewer 2: ```suggestion
This is the first release to follow our new, more predictable, more rigorous release process. It's outlined in the [release issue template](https://github.com/ipfs/js-ipfs/blob/master/doc/RELEASE_ISSUE_TEMPLATE.md), but in short we have defined steps, with an additional period for internal & external testing and gathering feedback on the changes.
```

> It's also the first release to ship with lockfiles for `yarn` and `npm`. This is to ensure that you only get the dependcies that we have validated with CI, and not untested code that may have been released with breaking changes since CI last ran on the release candidate.

Reviewer 2: ```suggestion
It's also the first release to ship with lockfiles for `yarn` and `npm`. This is to ensure that you only get the dependencies that we have validated with CI, and not untested code that may have been released with breaking changes since CI last ran on the release candidate.
```
